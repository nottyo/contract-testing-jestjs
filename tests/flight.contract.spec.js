import { flightRouteSchema } from '@/support/flight_route_schema'
import HttpStatus from 'http-status-codes'
import { FlightAPI } from '@/support/flight_api'
import { matchers } from 'jest-json-schema'
let flightAPI

describe('Flight API Contract Test', () => {
    beforeAll(() => {
        flightAPI = new FlightAPI()
        expect.extend(matchers)
    })

    test(`should return data as its spec`, async () => {
        // when
        const response = await flightAPI.getFlight('BKK', 'ICN')
        // then
        expect(response.status).toBe(HttpStatus.OK)
        expect(response.data).toMatchSchema(flightRouteSchema)
    })

    test(`response should matched with snapshots`, async () => {
        // when
        const response = await flightAPI.getFlight('BKK', 'ICN')
        // then
        expect(response.status).toBe(HttpStatus.OK)
        expect(response.data.status).toMatchSnapshot()
        response.data.data.forEach((flightData) => {
            expect(flightData).toMatchSnapshot({
                aircraft: expect.objectContaining({
                    manufacturer: expect.toBeOneOf(['Boeing', 'Airbus']),
                    model: expect.any(String)
                }),
                airline: expect.any(String),
                arrive: expect.stringMatching(/[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}\+[0-9]{2}:[0-9]{2}/),
                depart: expect.stringMatching(/[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{3}\+[0-9]{2}:[0-9]{2}/),
                flight: expect.objectContaining({
                    code: expect.any(String),
                    number: expect.any(Number)
                }),
                flightStatus: expect.toBeOneOf(['SCHEDULED', 'DELAYED', 'DIVERTED']),
                isDirectFlight: expect.any(Boolean)
            })
        })
    })
})