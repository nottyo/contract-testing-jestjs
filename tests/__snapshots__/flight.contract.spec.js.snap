// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flight API Contract Test response should matched with snapshots 1`] = `"success"`;

exports[`Flight API Contract Test response should matched with snapshots 2`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 3`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 4`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 5`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 6`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 7`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 8`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 9`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 10`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 11`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 12`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 13`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 14`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 15`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 16`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;

exports[`Flight API Contract Test response should matched with snapshots 17`] = `
Object {
  "aircraft": ObjectContaining {
    "manufacturer": toBeOneOf<Boeing,Airbus>,
    "model": Any<String>,
  },
  "airline": Any<String>,
  "arrive": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "depart": StringMatching /\\[0-9\\]\\{4\\}-\\[0-9\\]\\{2\\}-\\[0-9\\]\\{2\\}T\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}\\.\\[0-9\\]\\{3\\}\\\\\\+\\[0-9\\]\\{2\\}:\\[0-9\\]\\{2\\}/,
  "flight": ObjectContaining {
    "code": Any<String>,
    "number": Any<Number>,
  },
  "flightStatus": toBeOneOf<SCHEDULED,DELAYED,DIVERTED>,
  "isDirectFlight": Any<Boolean>,
}
`;
